src/arithmetic/arithmetic_expression.h

```diff
@@ -53,10 +53,11 @@ typedef struct {
 	union {
 		AR_FuncDesc *f;
 		AggCtx *agg_func;
-	};                            /* Operation to perform on children. */
-	const char *func_name;        /* Name of function. */
-	int child_count;              /* Number of children. */
-	struct AR_ExpNode **children; /* Child nodes. */
+	};                              /* Operation to perform on children. */
+	const char *func_name;          /* Name of function. */
+	int child_count;                /* Number of children. */
+	struct AR_ExpNode **children;   /* Child nodes. */
+    bool distinct;                  /* Rather or not this op performs distinction. */
```

Comment:
I don't feel like this member should be necessary. If it has to exist anywhere, it would be preferable to place it in the `AggCtx`.

Reply:
I need a way to differentiate between aggregated ExpOpNode and  none aggregated ExpOp, see ARExp_Clone.